packages/contracts/src/addresses.ts

```diff
@@ -187,3 +191,146 @@ export const getCheckBalanceLowAddress = () => {
+    const aBigInt = BigInt(a)
+    const bBigInt = BigInt(b)
+    return aBigInt < bBigInt ? -1 : aBigInt > bBigInt ? 1 : 0
+  })
+}
+
+export const getGnosisSafeInitializerData = (
+  owners: Array<string>,
+  threshold: number
+): string => {
+  if (owners.length === 0) {
+    throw new Error(
+      "Sphinx: You must have at least one owner in your 'owners' array."
+    )
+  }
+  if (threshold === 0) {
+    throw new Error(
+      "Sphinx: You must set your 'threshold' to a value greater than 0."
+    )
+  }
+
+  // Sort the owner addresses
+  const sortedOwners = sortHexStrings(owners)
+
+  const sphinxModuleProxyFactoryAddress = getSphinxModuleProxyFactoryAddress()
+
+  const sphinxModuleProxyFactory = new ethers.Contract(
+    sphinxModuleProxyFactoryAddress,
+    SphinxModuleProxyFactoryABI
+  )
+
+  // Encode the data for deploying the Sphinx Module
+  const encodedDeployModuleCall =
+    sphinxModuleProxyFactory.interface.encodeFunctionData(
+      'deploySphinxModuleProxyFromSafe',
+      [ethers.ZeroHash]
+    )
+
+  // Encode the data in a format for MultiSend
+  const deployModuleMultiSendData = ethers.solidityPacked(
+    ['uint8', 'address', 'uint', 'uint', 'bytes'],
+    [
+      Operation.Call,
+      sphinxModuleProxyFactoryAddress,
+      0,
+      ethers.getBytes(encodedDeployModuleCall).length,
+      encodedDeployModuleCall,
+    ]
+  )
+
+  // Similar encoding for enabling the Sphinx Module
+  const encodedEnableModuleCall =
+    sphinxModuleProxyFactory.interface.encodeFunctionData(
+      'enableSphinxModuleProxyFromSafe',
+      [ethers.ZeroHash]
+    )
+
+  const enableModuleMultiSendData = ethers.solidityPacked(
+    ['uint8', 'address', 'uint', 'uint', 'bytes'],
+    [
+      Operation.DelegateCall,
+      sphinxModuleProxyFactoryAddress,
+      0,
+      ethers.getBytes(encodedEnableModuleCall).length,
+      encodedEnableModuleCall,
+    ]
+  )
+
+  // Encode the entire MultiSend data
+  const multiSend = new ethers.Contract(
+    getMultiSendAddress(),
+    MultiSendArtifact.abi
+  )
+  const multiSendData = multiSend.interface.encodeFunctionData('multiSend', [
+    ethers.concat([deployModuleMultiSendData, enableModuleMultiSendData]),
+  ])
+
+  // Encode the call to the Gnosis Safe's `setup` function
+  const gnosisSafe = new ethers.Contract(
+    getGnosisSafeSingletonAddress(),
+    GnosisSafeArtifact.abi
+  )
+  const safeInitializerData = gnosisSafe.interface.encodeFunctionData('setup', [
+    sortedOwners,
+    threshold,
+    getMultiSendAddress(),
+    multiSendData,
+    // This is the default fallback handler used by Gnosis Safe during their
+    // standard deployments.
+    getCompatibilityFallbackHandlerAddress(),
+    // The following fields are for specifying an optional payment as part of the
+    // deployment. We don't use them.
+    ethers.ZeroAddress,
+    0,
+    ethers.ZeroAddress,
+  ])
+
+  return safeInitializerData
+}
+
+export const getGnosisSafeProxyAddress = (
```

Comment:
Since project creation occurs entirely via the website, we need a method for calculating the safe address in JavaScript to store in the DB. This function was originally in the test files, I pulled it into the contracts package and added some tests for it so we can use it in production.